packages/ui/src/components/TableV2/HeaderCell.tsx

```diff
@@ -0,0 +1,132 @@
+import styled from '@emotion/styled'
+import type { ReactNode } from 'react'
+import { Icon } from '../Icon'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+
+const StyledIconContainer = styled(Stack)`
+  color: ${({ theme }) => theme.colors.neutral.textWeak};
+
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+  }
+
+  &[role*='button'] {
+    cursor: pointer;
+    user-select: none;
+  }
+`
+
+const SortIcon = ({ order }: { order?: 'ascending' | 'descending' }) => (
+  <StyledIconContainer>
+    <Icon
+      name="arrow-up"
+      size={10}
+      color={order === 'ascending' ? 'primary' : 'neutral'}
+      prominence="weak"
+    />
+    <Icon
+      name="arrow-down"
+      size={10}
+      color={order === 'descending' ? 'primary' : 'neutral'}
+      prominence="weak"
+    />
+  </StyledIconContainer>
+)
+
+type StyledHeaderCellProps = Pick<
+  HeaderCellProps,
+  'width' | 'maxWidth' | 'minWidth'
+>
+const StyledHeaderCell = styled('th', {
+  shouldForwardProp: prop => !['width', 'maxWidth', 'minWidth'].includes(prop),
+})<StyledHeaderCellProps>`
+  ${({ width, maxWidth, minWidth }) => `
+    ${width ? `width: ${width};` : ''}
+    ${maxWidth ? `max-width: ${maxWidth};` : ''}
+    ${minWidth ? `min-width: ${minWidth};` : ''}
+  `}
+  padding: ${({ theme }) => theme.space['1']};
+
+  &[role*='button'] {
+    cursor: pointer;
+    user-select: none;
+  }
+`
+
+const StyledText = styled(Text)`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: ${({ theme }) => theme.space['1']};
+`
+
+type HeaderCellProps = {
+  children: ReactNode
+  className?: string
+  isOrdered?: boolean
+  orderDirection?: 'asc' | 'desc' | 'none'
+  onOrder?: (newOrder: 'asc' | 'desc') => void
+  width?: string
+  minWidth?: string
+  maxWidth?: string
+}
+
+export const HeaderCell = ({
+  children,
+  className,
+  isOrdered,
+  onOrder,
+  orderDirection,
+  width,
+  maxWidth,
+  minWidth,
+}: HeaderCellProps) => {
+  let order: undefined | 'ascending' | 'descending'
+  if (isOrdered && orderDirection === 'asc') {
+    order = 'ascending'
+  } else if (isOrdered && orderDirection === 'desc') {
+    order = 'descending'
+  }
+
+  const handleOrder = onOrder
+    ? () => onOrder(order === 'ascending' ? 'desc' : 'asc')
+    : undefined
+
+  return (
+    <StyledHeaderCell
+      className={className}
+      onClick={handleOrder}
+      onKeyDown={
+        handleOrder
+          ? event => {
+              if (event.key === ' ' || event.key === 'Enter') {
```

Comment:
```suggestion
              if (event.key === ' ') {
```
![Screenshot 2023-05-03 at 10 32 43](https://user-images.githubusercontent.com/15812968/235867646-c28b297e-be37-4e75-a84f-932bd40eb1da.png)

Here is a nice tool to get the value of each keys: https://www.toptal.com/developers/keycode